server/sonar-server/src/main/java/org/sonar/server/platform/monitoring/CeDatabaseMonitor.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.platform.monitoring;
+
+import java.util.LinkedHashMap;
+import org.sonar.process.ProcessId;
+import org.sonar.process.jmx.CeDatabaseMBean;
+import org.sonar.process.jmx.JmxConnection;
+import org.sonar.process.jmx.JmxConnector;
+
+public class CeDatabaseMonitor implements Monitor {
+
+  private final JmxConnector jmxConnector;
+
+  public CeDatabaseMonitor(JmxConnector jmxConnector) {
+    this.jmxConnector = jmxConnector;
+  }
+
+  @Override
+  public String name() {
+    return "Compute Engine Database";
+  }
+
+  @Override
+  public LinkedHashMap<String, Object> attributes() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The return type of this method should be an interface such as "Map" rather than the implementation "LinkedHashMap". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1319)